spec/Section 3 -- Type System.md

```diff
@@ -1340,14 +1359,283 @@ Union type extensions have the potential to be invalid if incorrectly defined.
+that is only valid for input, and another member field that's only valid for
+output, then it is not a valid Tagged type.
+
+A member field of a Tagged type may be of any valid GraphQL type, including
+Scalar, Enum, Input Object type, Object type, an Interface, a Union, another
+Tagged type, or any wrapping type (e.g. list, non-null, or any combination
+thereof) whose underlying base type is one of those seven.
+
+Selecting all the member fields of our `PackQuantity` type:
+
+```graphql example
+{
+  volumeInMetresCubed
+  numberOfBoxes
+  description
+}
+```
+
+Could yield one of the following results:
+
+- `{ "volumeInMetresCubed": 3.8 }`
+- `{ "numberOfBoxes": 27 }`
+- `{ "description": "4 large boxes, 7 medium boxes, and 16 small boxes" }`
+
+Valid queries must supply a nested field set for a field that returns a Tagged
+type, so for this schema:
+
+```graphql example
+type Query {
+  packQuantity: PackQuantity
+}
+```
+
+This query is not valid:
+
+```graphql counter-example
+{
+  packQuantity
+}
+```
+
+However, this query is valid:
+
+```graphql example
+{
+  packQuantity {
+    numberOfBoxes
+  }
+}
+```
+
+And may yield one of the following results:
+
+- `{ "numberOfBoxes": 27 }`
+- `{}`
+
+**Field Ordering**
+
+Like with Object types, when querying a Tagged type, the resulting mapping of
+fields are conceptually ordered in the same order in which they were
+encountered during query execution, excluding fragments for which the type does
+not apply and fields or fragments that are skipped via `@skip` or `@include`
+directives, and fields which are neither the single matched member field nor
+the `__typename` introspection field. This ordering is correctly produced when
+using the {CollectFields()} algorithm.
+
+**Result Coercion**
+
+Determining the result of coercing an object or tagged type is the heart of the
+GraphQL executor, so this is covered in that section of the spec.
+
+Note: only Tagged types for which {IsOutputType(type)} returns {true} may be
+used in output.
+
+**Input Coercion**
+
+Only Tagged types for which {IsInputType(type)} returns {true} may be used as
+input.
+
+The value for an input Tagged type should be an input object literal or an
+unordered map supplied by a variable, otherwise a query error must be thrown.
+In either case, the input object literal or unordered map must not contain any
+entries with names not defined by a member field of this Tagged type, otherwise
+an error must be thrown. The input object literal or unordered map must contain
+exactly one member field, otherwise an error must be thrown.
+
+The result of coercion is an unordered map with an entry for exactly one member
+field both defined by the tagged type and present in the input. The resulting
+map is constructed with the following rules:
+
+* If more than one field is defined in the input, an error must be thrown.
+
+* For the single field defined in the input, if the value is {null} and the
```

Comment:
This rule seems like it ought to be covered by a general "the input to that single field must match the fields' type"?